superset/db_engine_specs/postgres.py

```diff
@@ -205,17 +185,19 @@ def get_schema_from_engine_params(
         to determine the schema for a non-qualified table in a query. In cases like
         that we raise an exception.
         """
-        options = parse_options(connect_args)
-        if search_path := options.get("search_path"):
-            schemas = search_path.split(",")
-            if len(schemas) > 1:
-                raise Exception(
-                    "Multiple schemas are configured in the search path, which means "
-                    "Superset is unable to determine the schema of unqualified table "
-                    "names and enforce permissions."
-                )
-            return schemas[0]
-
+        options = re.split(r"-c\s?", connect_args.get("options", ""))
```

Comment:
I think we can actually get rid of the `get_schema_from_engine_params` method here, since Redshift doesn't support passing `search_path` as far as I could see.